lib/sbom.ex

```diff
@@ -50,6 +60,14 @@ defmodule SBoM do
     Mix.Dep.ok?(dep) || Mix.Dep.compilable?(dep)
   end
 
+  defp component_from_project(opts, type) do
+    %{
+      type: type,
+      name: to_string(opts[:app]),
```

Comment:
The example project I'm working with is an umbrella project, which doesn't have `app` defined in the root `mix.exs`, and `version` is configured statically as `1.0.0`.

This is what I end up with:

```
{
  "bomFormat": "CycloneDX",
  "specVersion": "1.6",
  "serialNumber": "urn:uuid:e05757dc-d8e1-4e91-9c40-213b694bd93c",
  "metadata": {
    "timestamp": "2025-05-26T11:11:48Z",
    "tools": [
      {
        "name": "SBoM Mix task for Elixir",
        "version": "0.6.3",
        "hashes": [],
        "externalReferences": []
      }
    ],
    "component": {
      "type": "application",
      "authors": [],
      "name": "",
      "version": "1.0.0",
      "tags": [],
      "omniborId": [],
      "swhid": []
    },
    "lifecycles": []
  },
```

Reply:
I don't know if there's any good standard way to find the application name in this case, so maybe we would need a way to provide custom values for all these properties... 

More CLI options?

Reply:
I think that would work for me 👍 

Github actions (`GITHUB_WORKSPACE`) also checks out stuff in `org/repo` afaik.

Reply:
Ok, I implemented this as a hack, it is not very pretty, but a better solution will have to wait until the rewrite of this tool.

BTW, I think you can just add `app: :some_name` to give assign a name to your umbrella Mix project. The contents of the Mix project's configuration is very flexible, tasks are supposed to ignore stuff they don't need.